packages/prerender/src/internal.ts

```diff
@@ -0,0 +1,50 @@
+import fs from 'fs'
+import path from 'path'
+
+import fetch from 'node-fetch'
+
+import type { AuthContextInterface } from '@redwoodjs/auth'
+import { getConfig, getPaths } from '@redwoodjs/internal'
+
+const INDEX_FILE = path.join(getPaths().web.dist, '/index.html')
+const DEFAULT_INDEX = path.join(getPaths().web.dist, '/defaultIndex.html')
+
+export const getRootHtmlPath = () => {
+  if (fs.existsSync(DEFAULT_INDEX)) {
+    return DEFAULT_INDEX
+  } else {
+    return INDEX_FILE
+  }
+}
+
+export const registerShims = () => {
+  global.__REDWOOD__API_PROXY_PATH = getConfig().web.apiProxyPath
+
+  global.__REDWOOD__USE_AUTH = () =>
+    ({
+      loading: true, // this should play nicely if the app waits for auth stuff to comeback first before render
+      isAuthenticated: false,
+    } as AuthContextInterface) // we only need a parital AuthContextInterface for prerender
+
+  global.__REDWOOD__PRERENDERING = true
+
+  // Let routes auto loader plugin know
+  process.env.__REDWOOD__PRERENDERING = 'true'
+
+  // We do this to avoid increasing the apollo client bundle size
```

Comment:
Isn't this run in the nodejs context? Curious about why it would increase the apollo client bundle size?

Reply:
If we don't shim fetch here, you would need to change the Apollo client to use cross-fetch or equivalent, and ship it with the bundle.

Reply:
Ok, I think the comment could use a bit of clarification about what the alternative is that it's trying to avoid.

Reply:
✅